### ISSUE TITLE
Number of people who have access to sharing this document

### ISSUE THREAD

**Issue by cyberbaloo** [suitenumerique/docs#1998]
## Bug Report

**Problematic behavior**
The visual information clearly tells us that the document is shared with 5333 people. This information is not conveyed by screen readers.

**Expected behavior/code**
It would be preferable to display the number of users who have access to this document.

Either modify the `aria-label` attribute to complete it with the missing information, or add an `aria-describedby` attribute pointing to the element that has this information, if the button name must remain short.

**Steps to Reproduce**
1. Log in to the document
2. Enable the screen reader
3. Navigate with the keyboard to the share button
4. And then the bug happens!

**Additional context/Screenshots**

<img width="852" height="221" alt="Image" src="https://github.com/user-attachments/assets/9cb6e4e5-7732-4329-a996-cc2e9e5b3b0f" />

**Critères RGAA**

Critère 6.1 : [[Chaque lien est-il explicite (hors cas particuliers) ?](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#6.1)](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#6.1)

**Impact**
A sighted user directly perceives the information while a screen reader user does not have access to this information. They don't have it even before opening the modal.
